DSpace/DSpace
# Improve REST Error message for specific errors

## References
* Fixes https://github.com/DSpace/DSpace/issues/3101

## Description
This PR delivers us a framework in which we can supply custom Exceptions with a message key and implement the `TranslatabeException` interface. When adding this Exception to the correct method in `DSpaceApiExceptionControllerAdvice`, the message returned in the Response will be translated based on the key that the Exception had. This message key will be retrieved from the appropriate Message.properties file depending on the locale of the Request.
The Locale of the Request can be set with the "Accept-Language" header.

## Instructions for Reviewers

List of changes in this PR:
* Added `TranslatableException` interface which allows us to create Exceptions with a message key. This message key will be looked up in the Messages(-LOCALE).properties file and the String in there will be used for the Response
* Added support for this in the `DSpaceApiExceptionControllerAdvice`
* Added tests to prove that it works

How to test this PR:
* Build and deploy the code locally
* Try to create an EPerson without a name
* Verify that you get a proper error message in the Response
* Add the key for this Exception to a Messages_pl.properties file and repeat the same request with "Accept-Languages" header set to "pl" ( Keep in mind that the `webui.supported.locales` property will need to contain the pl locale as well)
* Verify that the request with the Accept Languages header now returns the polish error message in the Response

## Checklist
